dlib/dnn/tensor.h

```diff
@@ -205,7 +205,7 @@ namespace dlib
         long m_k;
         long m_nr;
         long m_nc;
-        long m_size; // always equal to m_n*m_k*m_nr*m_nc
```

Comment:
Changing anything from signed to unsigned is going to create hard to find bugs.  It would be better to use long long here.

Reply:
Fixed.

Reply:
You mean to use long long right?

Reply:
It's probably fine to just change them all to long long. Probably even preferable since there are many places where people will multiply these things together.

Reply:
Changed all to long long.